bundles/org.eclipse.passage.loc.workbench/src/org/eclipse/passage/loc/internal/workbench/SelectRequest.java

```diff
@@ -0,0 +1,88 @@
+/*******************************************************************************
+ * Copyright (c) 2020 ArSysOp
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ArSysOp - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.loc.internal.workbench;
+
+import java.util.function.Supplier;
+
+import org.eclipse.passage.loc.jface.dialogs.Appearance;
+
+/**
+ * Encapsulates the information required to select the classifier
+ * 
+ * @since 0.6
+ *
+ * @param <R> target type to be selected
+ */
+public final class SelectRequest<R> {
+
+	private final Class<R> target;
+	private final String domain;
+	private final Supplier<Iterable<R>> input;
+	private final Appearance appearance;
+
+	/**
+	 * Creates a {@link SelectRequest} instance for target classifier that belongs
+	 * to a given domain, also specifies the input to select from and the dialog
+	 * appearance
+	 * 
+	 * @param target     the type of object to be selected, must not be
+	 *                   <code>null</code>
+	 * @param domain     the domain that target belongs to, must not be
+	 *                   <code>null</code>
+	 * @param input      the input to select from, must not be <code>null</code>
+	 * @param appearance the appearance of UI dialog, must not be <code>null</code>
+	 */
+	public SelectRequest(Class<R> target, String domain, Supplier<Iterable<R>> input, Appearance appearance) {
+		this.target = target;
+		this.domain = domain;
+		this.input = input;
+		this.appearance = appearance;
+	}
```

Comment:
If these guys are demanded to be not *null*, can we add `Objects.requireNotNull` checks?

Reply:
Yes, sure